org.eclipse.jdt.ui/core extension/org/eclipse/jdt/internal/corext/codemanipulation/AddImportsOperation.java

```diff
@@ -354,15 +348,12 @@ private TextEdit evaluateEdits(CompilationUnit root, ImportRewrite importRewrite
 		}
 		IJavaSearchScope searchScope= SearchEngine.createJavaSearchScope(new IJavaElement[] { fCompilationUnit.getJavaProject() });
 
-		TypeNameMatch[] types= findAllTypes(simpleName, searchScope, nameNode, new SubProgressMonitor(monitor, 1));
+		TypeNameMatch[] types= findAllTypes(simpleName, searchScope, nameNode, subMonitor.split(1));
 		if (types.length == 0) {
 			fStatus= JavaUIStatus.createError(IStatus.ERROR, Messages.format(CodeGenerationMessages.AddImportsOperation_error_notresolved_message, BasicElementLabels.getJavaElementName(simpleName)), null);
 			return null;
 		}
 
-		if (monitor.isCanceled()) {
```

Comment:
Any reason this is no longer valid?

Reply:
From what I understand, one of the features of `SubMonitor` is to provide an abstraction where the caller doesn't have to do explicit cancellation handling.

From the doc of `SubMonitor#split`:

> This method is much like newChild, but it will additionally check for cancellation and will throw an OperationCanceledException if the monitor has been cancelled.

This change moves the cancellation check (implicit check by calling `SubMonitor#split`) above the if block, but in my mind that doesn't really make a difference as the if block is just a guard clause that doesn't do a costly computation.

Reply:
Seems reasonable.